rust/src/libp2p_bridge.rs

```diff
@@ -17,139 +25,193 @@ static mut SWARM_STATE: Option<libp2p::swarm::Swarm<Behaviour>> = None;
 #[no_mangle]
-pub fn create_and_run_network(network_id: u32, zig_handler: u64, self_port: i32, connect_port: i32) {
-
-    let rt = Builder::new_current_thread()
-        .enable_all()
-        .build()
-        .unwrap();
-
-        rt.block_on(async move {
-            let mut p2p_net = Network::new(network_id, zig_handler);
-           p2p_net.start_network(self_port, connect_port).await;
-           p2p_net.run_eventloop().await;
-
-        });
+pub unsafe fn create_and_run_network(
+    network_id: u32,
+    zig_handler: u64,
+    listen_addresses: *const c_char,
+    connect_addresses: *const c_char,
+) {
+    let listen_multiaddrs = std::ffi::CStr::from_ptr(listen_addresses)
+        .to_string_lossy()
+        .split(",")
+        .map(|addr| addr.parse::<Multiaddr>().expect("Invalid multiaddress"))
+        .collect::<Vec<_>>();
+
+    let connect_multiaddrs = std::ffi::CStr::from_ptr(connect_addresses)
+        .to_string_lossy()
+        .split(",")
+        .filter(|s| !s.trim().is_empty()) // filter out empty strings because connect_addresses can be empty
+        .map(|addr| addr.parse::<Multiaddr>().expect("Invalid multiaddress"))
+        .collect::<Vec<_>>();
+
+    releaseAddresses(zig_handler, listen_addresses, connect_addresses);
+
+    let rt = Builder::new_current_thread().enable_all().build().unwrap();
+
+    rt.block_on(async move {
+        let mut p2p_net = Network::new(network_id, zig_handler);
+        p2p_net
+            .start_network(listen_multiaddrs, connect_multiaddrs)
+            .await;
+        p2p_net.run_eventloop().await;
+    });
 }
 
 /// # Safety
 /// The caller must ensure that `message_str` points to valid memory of `message_len` bytes.
 #[no_mangle]
 #[allow(clippy::missing_safety_doc)]
-pub unsafe fn publish_msg_to_rust_bridge(network_id:u32,topic_id: u32, message_str: *const u8, message_len: usize){
-        let message_slice = std::slice::from_raw_parts(message_str, message_len);
-        println!("rustbridge-{network_id}:: publishing message s={:?}",message_slice);
-        let message_data = message_slice.to_vec();
-
-        // TODO: get the topic mapping from topic_id
-        let topic = match topic_id {
-            0 => gossipsub::IdentTopic::new("block"),
-            1 => gossipsub::IdentTopic::new("vote"),
-            unknown_id => {
-                println!("Invalid topic_id: {unknown_id}");
-                return;
-            },
-        };
-        
-         #[allow(static_mut_refs)]
-         let swarm = if network_id < 1 {unsafe {SWARM_STATE.as_mut().unwrap()}} else {unsafe {SWARM_STATE1.as_mut().unwrap()}};
-        // let mut swarm = unsafe {SWARM_STATE.as_mut().unwrap()};
-        if let Err(e) = swarm.behaviour_mut().gossipsub
-                    .publish(topic.clone(), message_data){
-                    println!("Publish error: {e:?}");
-                }
+pub unsafe fn publish_msg_to_rust_bridge(
+    network_id: u32,
+    topic_id: u32,
```

Comment:
```suggestion
    topic_id: u32,
```
@noopur23 also create another issue and assign to Kai to replace topic_id with actual topic string so we don't have to do the topic_id to topic conversion below

similarly in call, directly pass topic string to zig where zig does the topic string to topic mapping instead of topic_id again being an intermediary